backend/src/main/java/com/agh/introwertycznelosie/config/MajorController.java

```diff
@@ -17,20 +19,54 @@
     @Autowired
     MajorService majorService;
 
+    @Autowired
+    FacultyService facultyService;
+
     @GetMapping(value="/newest-majors", produces = MediaType.APPLICATION_JSON_VALUE)
     public List<Major> getMajors() {
         return majorService.get();
     }
 
     @PostMapping("/new-major")
     public ResponseEntity<HttpStatus> postNewMajor(@RequestBody Major major) {
+        Faculty faculty = facultyService.findByAcronym(major.getFaculty().getAcronym());
+        if(faculty==null)
+        {
+            faculty = facultyService.save(major.getFaculty());
+        }
+        major.setFaculty(faculty);
         majorService.save(major);
         return ResponseEntity.ok(HttpStatus.OK);
     }
 
+    @PutMapping("edit-major/{id}")
+    public Major updateMajor(@RequestBody Major major, @PathVariable Long id) {
```

Comment:
Może editMajor skoro linkujemy do edit-major ? Chyba, że taką przyjmujemy konwencję też w innych edycjach to się nie kłócę

Reply:
Tak było z updateRoom IL-28 na którym się wzorowałem